megamek/src/megamek/client/bot/princess/BasicPathRanker.java

```diff
@@ -387,53 +389,15 @@ protected double calculateHerdingMod(Coords friendsCoords, MovePath path) {
         return herdingMod;
     }
 
-    protected double calculateFacingMod(Entity movingUnit, Game game, final MovePath path, @Nullable Coords closestPosition) {
+    protected double calculateFacingMod(Entity movingUnit, Game game, final MovePath path,
+          @Nullable Coords enemyMedianPosition, @Nullable Coords closestEnemyPosition) {
         // limit to the 5 closest enemies
-        int facingDiff = getFacingDiff(movingUnit, game, closestPosition, path);
-        double facingMod = Math.max(0.0, 50 * (facingDiff - 1));
-        logger.trace("facing mod [-{} = max(0, 50 * ({}) - 1)]", facingMod, facingDiff);
-        return facingMod;
-    }
-
-    private int getFacingDiff(Entity movingUnit, Game game, @Nullable Coords closestPosition, final MovePath path) {
-        Coords toFace = closestPosition == null ? game.getBoard().getCenter() : closestPosition;
-        int desiredFacing = (toFace.direction(movingUnit.getPosition()) + 3) % 6;
-        int currentFacing = path.getFinalFacing();
-        int facingDiff;
-        // -1 is bias towards facing left, 1 is bias towards facing right
-        int biasTowardsFacing = getBiasTowardsFacing(movingUnit);
-        desiredFacing -= biasTowardsFacing;
-        if (desiredFacing < 0) {
-            desiredFacing += 6;
-        }
-        if (currentFacing == desiredFacing) {
-            facingDiff = 0;
-        } else if ((currentFacing == ((desiredFacing + 1) % 6))
-                || (currentFacing == ((desiredFacing + 5) % 6))) {
-            facingDiff = 1;
-        } else if ((currentFacing == ((desiredFacing + 2) % 6))
-                || (currentFacing == ((desiredFacing + 4) % 6))) {
-            facingDiff = 2;
-        } else {
-            facingDiff = 3;
-        }
-        return facingDiff;
-    }
+        int facingDiff = facingDiffCalculator.getFacingDiff(movingUnit, path, game.getBoard().getCenter(),
+              enemyMedianPosition, closestEnemyPosition);
+        double facingMod = 50 * facingDiff;
```

Comment:
Since we utilize the "50" value in several places, it should be a constant or argument.